internal/git_config/git_config.go

```diff
@@ -134,3 +152,50 @@ func SetOption(logger *zerolog.Logger, folderPath, key string, value string) {
 		logger.Error().Err(err).Msgf("could not store %s=%s configuration for folder %s", key, value, folderPath)
 	}
 }
+
+func GitRepoFolderPath(folderPath string) (string, error) {
+	repo, err := git.PlainOpen(folderPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to open repository: %s %w", folderPath, err)
+	}
+
+	fsStorer, ok := repo.Storer.(*filesystem.Storage)
+	if !ok {
+		return "", fmt.Errorf("faild to get fs storage for: %s %w", folderPath, err)
+	}
+	repoPath := fsStorer.Filesystem().Root()
+	if repoPath == "" {
+		return "", errors.New("repository path is empty")
+	}
+
+	if !strings.HasSuffix(repoPath, ".git") {
+		repoPath = filepath.Join(repoPath, ".git")
+	}
+
+	return repoPath, nil
+}
+
+// DeleteEmptySnykSubsection This is a migration function to be executed if empty subsections exists
+func DeleteEmptySnykSubsection() error {
+	gitFolderPath, err := GitRepoFolderPath("")
```

Comment:
go-git treats empty folder as the current folder. which is annoying.

here I'm loading folderPath="" to reproduce the same behavior that we might get.